backend/vkt/src/main/java/fi/oph/vkt/api/IndexController.java

```diff
@@ -9,9 +13,42 @@
 @RequestMapping(value = "/")
 public class IndexController {
 
+  private static final int NONCE_BYTES = 32;
+  private SecureRandom secureRandom = new SecureRandom();
+
+  private static void addCSPHeaders(HttpServletResponse response, String nonce) {
+    StringBuilder sb = new StringBuilder();
+    sb.append("default-src 'none'; ");
+
+    sb.append("script-src 'self' 'nonce-");
+    sb.append(nonce);
+    sb.append("' 'strict-dynamic' https: http: 'unsafe-inline'; ");
+
+    sb.append("style-src 'self' 'nonce-");
+    sb.append(nonce);
+    sb.append("' 'unsafe-inline'; ");
+
+    sb.append("connect-src 'self'; ");
+    sb.append("img-src 'self'; ");
+    sb.append("font-src 'self'; ");
+    sb.append("base-uri 'self'; ");
+    sb.append("form-action 'self';");
+
+    response.addHeader("Content-Security-Policy", sb.toString());
+  }
+
+  private String getNonce() {
+    byte[] nonceBytes = new byte[NONCE_BYTES];
+    secureRandom.nextBytes(nonceBytes);
+
+    return Base64.getEncoder().encodeToString(nonceBytes);
```

Comment:
Onko tarve nimen omaan 32 bittiin? Itselle tuli ensimmäisenä mieleen `UUID.randomUUID().toString()`.

Reply:
32 tavua, ei bittiä. :) Jotain ohjeita noncen kompleksisuudelle löytyy, esim. täällä: https://content-security-policy.com/nonce/ 

`The nonce should have sufficient length, aim for at least 128 bits of entropy (32 hex characters, or about 24 base64 characters).`

Reply:
`UUID.randomUUID()` ei sinänsä taida olla kovin hyvä, kun siinä asetellaan alkupään tavuja tiettyyn staattiseen asentoon. Lisäksi UUID:n toString taitaa palauttaa heksaa, mutta tässä halutaan Base64-enkoodausta.